client/ayon_core/tools/sceneinventory/view.py

```diff
@@ -927,24 +947,37 @@ def update_all(self):
         self._update_containers_to_version(item_ids, version=-1)
 
     def _on_switch_to_versioned(self, item_ids):
-        containers_items_by_id = self._controller.get_container_items_by_id(
-            item_ids
-        )
+        # Get container items by ID
+        containers_items_by_id = self._controller.get_container_items_by_id(item_ids)
         repre_ids = {
             container_item.representation_id
             for container_item in containers_items_by_id.values()
         }
-        repre_info_by_id = self._controller.get_representation_info_items(
-            repre_ids
-        )
-        product_ids = {
-            repre_info.product_id
-            for repre_info in repre_info_by_id.values()
-            if repre_info.is_valid
+        # Extract project names and their corresponding representation IDs
+        project_name_to_repre_ids = {}
+        for container_item in containers_items_by_id.values():
+            project_name = container_item.project_name
+            repre_id = container_item.representation_id
+            if project_name not in project_name_to_repre_ids:
+                project_name_to_repre_ids[project_name] = set()
+            project_name_to_repre_ids[project_name].add(repre_id)
+
+        # Get representation info items by ID
+        repre_info_by_id = self._controller.get_representation_info_items(repre_ids)
```

Comment:
This likely would also need to run per project and receive a project name so we know which project to query in?

Reply:
@iLLiCiTiT this would only ever make sense if `get_representation_info_items` takes `project_name` too, right?

Meaning all the logic here would just run per project.

Reply:
I'm afraid so.